File: frontend/src/types/task.ts
export type ApiTaskStatus = 'pending' | 'preprocess_done' | 'impute_done' | 'batch_done' | 'eval_done' | 'error' | string

export type ApiTask = {
  id: number
  task_name: string
  status: ApiTaskStatus
  created_at: string
  updated_at: string
  error_message?: string | null
}

export type TaskCreateBody = {
  task_id: number
  task_name?: string
  preprocess_config: Record<string, unknown>
  imputation_config: Record<string, unknown>
  batch_config: Record<string, unknown>
  evaluation_config: Record<string, unknown>
}

File: backend/app/schemas/task.py
from __future__ import annotations

from typing import Any, Dict, Optional

from pydantic import BaseModel, Field


class TaskCreateRequest(BaseModel):
    task_id: int = Field(..., description="上传后返回的 task_id")
    task_name: Optional[str] = Field(None)

    preprocess_config: Dict[str, Any] = Field(default_factory=dict)
    imputation_config: Dict[str, Any] = Field(default_factory=dict)
    batch_config: Dict[str, Any] = Field(default_factory=dict)
    evaluation_config: Dict[str, Any] = Field(default_factory=dict)


class TaskResponse(BaseModel):
    id: int
    task_name: str
    status: str
    created_at: str
    updated_at: str
    error_message: Optional[str] = None


class EvaluationResponse(BaseModel):
    task_id: int
    pca_before: Dict[str, Any]
    pca_after: Dict[str, Any]
    pca_plot_path: str

